app/controllers/kudos_controller.rb

```diff
@@ -23,7 +23,7 @@ def create
         format.html do
           flash[:comment_notice] = ts("Thank you for leaving kudos!")
 
-          redirect_to request.referer and return
+          redirect_to request.referer || root_path and return
```

Comment:
I'm not sure root_path is a good fallback for kudos -- if we can't redirect someone back to the _exact_ referrer (which could be something like a specific chapter of the work they're kudosing or the work's kudos), we should at least redirect them to work.

Reply:
I reverted this change.